src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectResultExecutor.cs

```diff
@@ -142,7 +142,18 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 objectType,
                 result.Value);
 
-            var selectedFormatter = SelectFormatter(formatterContext, result.ContentTypes, formatters);
+            IOutputFormatter selectedFormatter = null;
+            try
+            {
+                selectedFormatter = SelectFormatter(formatterContext, result.ContentTypes, formatters);
+            }
+            catch (InvalidOperationException e)
+            {
+                Logger.InvalidAcceptHeader(formatterContext, e);
```

Comment:
The `Exception` logged here contains a less-useful message than the previous `FormatException`.